### PR TITLE
Adds `StrEnum` support for python3.11

### PR DESCRIPTION
**Pull request by sobolevn**
Since there are already a couple of alpha versions available, I think it might be a good idea to add support for some new python features.

For example, now subclasses of `StrEnum` will be treated as real `Enum`s!

### CODE REVIEW

File: test-data/unit/lib-stub/enum.pyi [hunk 1]
```diff
@@ -43,3 +43,8 @@ class IntFlag(int, Flag):
 
 class auto(IntFlag):
     value: Any
+
+
+# It is python-3.11+ only:
+class StrEnum(str, Enum):
+    def __new__(cls: Type[_T], value: int | _T) -> _T: ...
```
Comment (AlexWaygood) [inline test-data/unit/lib-stub/enum.pyi:50]: [Over at typeshed](https://github.com/python/typeshed/pull/6580) I corrected this to 

```python
    def __new__(cls: Type[_T], value: str | _T) -> _T: ...
```

Comment (sobolevn) [reply test-data/unit/lib-stub/enum.pyi]: Yes, sure. This is a typo. Thanks!

Comment (AlexWaygood) [reply test-data/unit/lib-stub/enum.pyi]: No problem!

**Comment by JelleZijlstra** [comment]
Did we turn off 3.11 CI completely? I don't see it in the list.

**Comment by 97littleleaf11** [comment]
Yeah we just turned if off. https://github.com/python/mypy/pull/12003#issuecomment-1015366539

**Pull request merged**